### DISCUSSION TITLE
Strange Translation/Jittering Artifact in Phase Field Model

### DISCUSSION THREAD

**Discussion by aclin4** [idaholab/moose#21972] [Q&A Modules: Phase field] [answered]
I've encountered some strange behavior with one of my phase field models that I was wondering if anyone else has come across before. It's a fairly standard 2-phase (1 order parameter) 3-species (2 concentration variables) WBM model. In the two-phase region of my model, I observe the expected phase separation behavior, but it seems that my model seems to jitter randomly, translating in the x-direction after separating.

Have anyone ever seen this before or have any idea of what could be causing this?

Thanks,
Albert
![small](https://user-images.githubusercontent.com/106271284/187502366-e1983dd7-9464-41e4-9217-7c7d13d2b01b.gif)

**Comment by GiudGiud** [comment, marked as answer]
@SudiptaBiswas @laagesen

**Reply by laagesen** [reply]
That is weird. What boundary conditions are you using for all the variables? The behavior of the psi=1 phase on the LHS seems a little weird and I'm wondering if there may be some inconsistency in the BCs.

Also, what are your solver settings? You are approaching a steady-state condition at the end, so the Transient Executioner may struggle to converge the problem as you approach the steady-state. If you've increased `nl_rel_tol` to deal with this, that could potentially cause problems. Usually if you want to run a problem all the way to steady-state, it's best to set an absolute solver tolerance using `nl_abs_tol = 1e-11` or something along those lines. There are fancier ways to do this in MOOSE, but that has usually worked well for me in the past.

**Reply by laagesen** [reply]
Also I assume the variable here is a non-conserved order parameter because it doesn't look like it is being conserved (volume fraction seems to increase at the end). But, for a coupled model that may indicate that there is a problem with volume conservation with one of your conserved variables. I would add a postprocessor to integrate each of your concentration variables at each time step and make sure they are being conserved.

**Reply by aclin4** [reply]
Hi Larry,

You're right that the order parameter shown here is non-conserved; it's just evolving the phase fraction to accommodate concentration-enriched and depleted phases. I've checked that the concentration variables are indeed conserved throughout the simulation.

For all the variables, I'm using periodic boundary conditions:
<pre>
[BCs]
  [./Periodic]
    [./all]
      variable = 'c_cr c_ni psi w_ni w_cr'
      auto_direction = 'x y'
    [../]
  [../]
[]
</pre>
which is probably why there is an additional interface that appears on the left side of the simulation cell.

As for the Executioner settings, it seems that all the parameters are already pretty close to what you've suggested.
<pre>
[Executioner]
  type = Transient
  scheme = bdf2
  solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
  petsc_options_value = 'lu      superlu_dist'
  line_search = 'none'

  nl_max_its = 20
  l_max_its = 20
  l_tol = 1e-12
  nl_rel_tol = 1e-10
  nl_abs_tol = 1e-12
  end_time = 1e15
  #dtmax = 1e6
  dtmin = 2e-20
  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 1e-3
    cutback_factor = 0.5
    growth_factor = 1.25
    optimal_iterations = 12
  [../]
[]
</pre>
I thought it may be due to the way I set up my mesh adaptivity:
<pre>
[Adaptivity]
  marker = combo
  max_h_level = 2
  [./Indicators]
    [./psiind]
      type = GradientJumpIndicator
      variable = psi
    [../]
    [./Niind]
      type = GradientJumpIndicator
      variable = c_ni
    [../]
    [./Crind]
      type = GradientJumpIndicator
      variable = c_cr
    [../]
    [./wNiind]
      type = GradientJumpIndicator
      variable = w_ni
    [../]
    [./wCrind]
      type = GradientJumpIndicator
      variable = w_cr
    [../]
  [../]
  [./Markers]
   [./combo]
     type = ComboMarker
     markers =  'psierror cnierror ccrerror wcrerror wnierror'
   [../]
    [./psierror]
      type = ErrorFractionMarker
      coarsen = 0.2
      indicator = psiind
      refine = 0.7
    [../]
    [./cnierror]
      type = ErrorFractionMarker
      coarsen = 0.2
      indicator = Niind
      refine = 0.7
    [../]
    [./ccrerror]
      type = ErrorFractionMarker
      coarsen = 0.2
      indicator = Crind
      refine = 0.7
    [../]
    [./wcrerror]
      type = ErrorFractionMarker
      coarsen = 0.2
      indicator = wCrind
      refine = 0.7
    [../]
    [./wnierror]
      type = ErrorFractionMarker
      coarsen = 0.2
      indicator = wNiind
      refine = 0.7
    [../]
  [../]
[]
</pre>

Does that seem reasonable to you?

Thanks,
Albert

**Reply by laagesen** [reply]
I think you might have too many different markers/indicators. It could be refinement/coarsening is happening too frequently based on different interfacial widths for different variables and how the settings interact with them. It could be that excessive coarsening/refinement is causing the interfaces to move when you are suddenly solving the governing equations on a different grid, especially if the amount of refinement changes a lot between time steps. I would try solving the problem without mesh refinement to see if that makes the problem go away. Also I'd make sure your concentration variables are conserved based on what I observed before about the volume fraction.
